packages/shared-core/src/filters.ts

```diff
@@ -302,11 +302,16 @@ export const runLuceneQuery = (docs: any[], query?: SearchQuery) => {
       docValue: string | number | null,
       testValue: { low: number; high: number }
     ) => {
+      if (docValue == null || docValue === "") {
+        return true
+      }
+      if (!isNaN(+docValue)) {
+        return +docValue < testValue.low || +docValue > testValue.high
+      }
+      // if the string is NaN, then assume date
```

Comment:
Could we actually check if it's a valid date ISO? Otherwise, it should not work and catch the error